app/recurring_job.go

```diff
@@ -750,7 +750,15 @@ func (job *Job) findARandomReadyNode(v *longhornclient.Volume) (string, error) {
 
 	var readyNodeList []string
 	for _, node := range nodeCollection.Data {
-		if readyConditionInterface, ok := node.Conditions[longhorn.NodeConditionTypeReady]; ok {
+		var readyConditionInterface map[string]interface{}
+		for i := range node.Conditions {
+			con := node.Conditions[i].(map[string]interface{})
+			if con["type"] == longhorn.NodeConditionTypeReady {
+				readyConditionInterface = con
+			}
+		}
+
+		if readyConditionInterface != nil {
```

Comment:
Not sure _why_ below need to perform json marshal/unmarshal, can't we directly use the `node.Conditions`?

Reply:
Hmm, need to check the generic json marshal/unmarshal logic.

Reply:
@jenting The reason is the Conditions' datatype is map[string]interface{}, so need to marshal here again.

Reply:
Thanks for the explanation.